docs/source/troubleshooting.rst

```diff
@@ -51,28 +51,41 @@ Collect Debug Information
 
    You can update pyQuil with ``pip`` using ``pip install pyquil --upgrade``. You can find
    the latest version available at
-   `our releases page <https://github.com/rigetti/pyquil/releases>`_ or
+   `the releases page <https://github.com/rigetti/pyquil/releases>`_ or
    `on PyPi <https://pypi.org/project/pyquil/>`_.
 
-2. If the error appears to be authentication-related, refer to the `QCS CLI documentation
-<https://docs.rigetti.com>`_.
 
-3. Run your script with debug logging enabled. If you're running a script, you can enable that
-   using an environment variable:
+2. pyQuil exposes a diagnostics report that prints helpful debugging information, including
+   whether you have connectivity to ``quilc``, ``QVM`` and access to QCS services. You can
+   use it by importing a function from the ``diagnostics`` module:
+
+   .. testcode:: version
+
+    from pyquil.diagnostics import get_report
+    print(get_report())
```

Comment:
This is useful; may also be useful to include a one-liner script or to define a script in `pyproject.toml` so it can be invoked from the shell instead of an interpreter